components/calendar/src/ixdtf.rs

```diff
@@ -25,8 +25,8 @@ pub enum ParseError {
     /// The IXDTF specifies an unknown calendar.
     UnknownCalendar,
     /// Expected a different calendar.
-    #[displaydoc("Expected calendar {0} but found calendar {1}")]
-    MismatchedCalendar(AnyCalendarKind, AnyCalendarKind),
+    #[displaydoc("Expected calendar {0:?} but found calendar {1:?}")]
+    MismatchedCalendar(CalendarAlgorithm, CalendarAlgorithm),
```

Comment:
Thought: I think `CalendarAlgorithm` is better than `AnyCalendarKind` here. I wonder if it wants to have its own enum; for example, maybe IXDTF wants to support calendars that CLDR doesn't need. At one point I thought about supporting `islamic-thirty` as an islamic calendar useful for arithmetic (every calendar has 30 days) but not useful for other purposes like conversion to ISO. I don't think CLDR would want this calendar, but IXDTF might.

Reply:
From RFC 9557:  

> A calendar is a set of rules defining how dates are counted and
   consumed by implementations.  The set of suffix values allowed for
   this suffix key is the set of values defined for the Unicode Calendar
   Identifier [TR35].